File: src/main/java/gui/Canvas.java
package gui;

import camera.Line2D;

import javax.swing.*;
import java.awt.*;
import java.util.ArrayList;

public class Canvas extends JPanel {
    private ArrayList<Line2D> lines;

    // FIXME: powinien brac Scene
    public Canvas(ArrayList<Line2D> lines) {
        this.lines = lines;
    }

    @Override
    protected void paintComponent(Graphics graphics) {
        super.paintComponent(graphics);

        // FIXME: scene.draw(graphics)
        for (Line2D line: lines) {
            line.draw(graphics);
        }
    }
}

File: src/main/kotlin/gui/Canvas.kt
package gui

import camera.Camera
import camera.Scene

import javax.swing.*
import java.awt.*

class Canvas (var scene: Scene, private val camera: Camera) : JPanel() {
    override fun paintComponent(graphics: Graphics) {
        super.paintComponent(graphics)
        super.setBackground(Color.decode("#dae1e7"))
        graphics.drawRect(0, 0, Scene.HEIGHT, Scene.WIDTH)
        DrawingContext(graphics, scene, camera).draw()
    }
}
